### PR TITLE
retdec-devel: update to 20211115

### PR DESCRIPTION
**Pull request by catap**
#### Description

This also switched retdec port to using OpenSSL 1.1.1l which is required to fix arm64 build.

Fixes: https://trac.macports.org/ticket/63902

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.6 20G165 x86_64
Xcode 13.1 13A1030d

macOS 12.0.1 21A559 arm64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Review by mascguy — APPROVED** [review]

**Comment by catap** [comment]
@mascguy can I ask you to merge this PR? I'm a maintainer of this port and it passes CI. Thanks!

**Comment by mascguy** [comment]
Looks like the CI jobs are still in progress. Once they're complete, I'll merge.

File: devel/retdec/Portfile
Comment (reneeotten) [inline devel/retdec/Portfile:52]: isn't this what `supported_archs` is for?

File: devel/retdec/Portfile
Comment (reneeotten) [inline devel/retdec/Portfile:68]: why? then just don't add it again below

File: devel/retdec/Portfile
Comment (catap) [reply devel/retdec/Portfile:52]: @reneeotten `supported_archs i386 ppc ppc64 x86_64` triggers to build dependencies whcih includes `graphviz` with `+universal` which is failed on https://trac.macports.org/ticket/63561 but it replaces tons of ports to `+universal` variant.

Comment (reneeotten) [reply devel/retdec/Portfile:54]: I don't see how that has anything to do with this. If you cannot build for `arm64` but it should build for all other architectures, I the line you mention seems correct. If that doesn't work there must be something else going on...

Comment (catap) [reply devel/retdec/Portfile:54]: @reneeotten let me try to fix `dav1d`...

Comment (reneeotten) [reply devel/retdec/Portfile:54]: I don't see why that would be necessary if you're effectively just removing `arm64` from the lists of supported_archs. If it doesn't work then, it shouldn't work now either on other architectures, unless I am misunderstanding something here... admittedly I don't fully understand how this universal builds stuff is done... What happens if you add `universal_variant  no`?

Comment (catap) [reply devel/retdec/Portfile:54]: @reneeotten it still tries to rebuild everything for `+universal`. I'm fixing a few trivial port, but I afraid that it may trigger something huge.

Comment (reneeotten) [reply devel/retdec/Portfile:54]: then I'd suggest to just comment in that Trac ticket and/or wait for someone who understands this universal stuff to stop by at this PR and educate us (or at least me) on what's going on

Comment (catap) [reply devel/retdec/Portfile:54]: @reneeotten this is expected to be honest. M1 can run both x86_64 and arm64 application, and if application can't be build for arm64 but can be built for x86_64 => it tries to build this.

Is need to rebuild all dependencies with `+universal` which isn't `noarch` and if it can => just build x86_64 version.

Comment (catap) [reply devel/retdec/Portfile:54]: Ok, I gave up. It tries to build `rust` / `cargo` as `+universal` to build some library... and `rust` port defently not something which can be made `+universal` by some simple hack.

**Comment by catap** [comment]
From another hand, as soon as the new version is released, this hack won't be need anymore.

**Comment by catap** [comment]
From another hand, as soon as the new version is released, this hack won't be need anymore.

@reneeotten have you got anything in mind that prevent this to be merged?

**Comment by catap** [comment]
@reneeotten / @mascguy I figured out the better way to fix arm64 build: updated OpenSSL version to which one is supported arm64 build :)

Thus, my patch to OpenSSL 3 support was merged to master => no need to use it anymore.

File: devel/retdec/files/patch-openssl-v40.diff
Comment (mascguy) [inline devel/retdec/files/patch-openssl-v40.diff:46]: Kirill, does this have to exactly match our OpenSSL 1.1 port?

Comment (catap) [reply devel/retdec/files/patch-openssl-v40.diff:46]: @mascguy not at all. To be `1.1.x` is enough. If I haven't use OpenSSL PG, it may have OpenSSL 3.0 by dependencies inside `/opt/local/include/openssl` and it brokes build because it is linked agains embeded OpenSSL but uses headers from system one.

From another hand, embeded OpenSSL too old to support Apple's arm64 :)

=> simple upgrade hardcoded version seems a nice and easy hack. Much better than cross build via universal varian.

Thus, I've raised in upstream an idea to release a new version which included my patch to support OpenSSL 3.0; and I hope soon it happened.

**Comment by catap** [comment]
@mascguy I can make one more rebase and run one more loop of CI whcih will block it for next couple of hours.

Or we can merge it.

Anyway, I'm maintaner of this port and I'm using it :) here I feel that I can ask you just to merge it.

**Pull request merged**
